### ISSUE TITLE
SQL backend feature request

### ISSUE THREAD

**Issue by teqnodude** [glftpd/glftpd#37]
I wonder if the developers have ever thought about making logs and userdb backed by SQL as an option. This would enable all kinds of things if done correctly like shared userdb, easier overview of log files and especially if you are an admin of many sites where each log entry is tagged with sitename, better statistical overview, internal/external php based dashboard and administration of users. It would make it far easier to do a backup since you could have the database off site on a dedicated database server where the only thing you would require is a backup of glftpd.conf and certificate to restore function of the site after a reinstall or move to another server. As I see it the possibilities are almost endless if the glftpd server would have the ability to use SQL for both userdb and logs.

**Comment by terra-dude** [comment]
I like the idea, about time **gl** enters the 21st century :-) 

SQL to store data (optinal admin interfaces) / manage userdb, etc... 
And would be easier to use across different servers/devices.

**Comment by PatriotM** [comment]
that would be a great feature.

the dirlog is also not up to date with today's time. many folders make the dirlog slow. in combination with a script that checks the dirlog for duplicate it sometimes takes a long time until the folder is created.

the same for the other log files. i also think that sql would be a real performance boosting feature.

**Comment by sachaz** [comment]
Yeah welcome to the present, Mariadb as backend to play with could be very nice ! Users and their history, directories, all site commands history, metatags, etc... And maybe a redis cache for a fast site search ?

**Comment by glftpd** [comment]
If someone is willing to implement it as a plugin then just msg Hujer on efnet. For starters i can provide list of api calls for the dirlog replacement (fairly simple) and we can do other parts later.
